### DISCUSSION TITLE
Path toString then split the string and convert to the path again

### DISCUSSION THREAD

**Discussion by zillurbmb51** [nextflow-io/nextflow#3333] [Q&A]
Hi, I have a process that output multiple files. Such as:
`[chr1, /path/to/chr1_chunk1.csv.bed, /path/to/chr1_chunk1.csv.bim, /path/to/chr1_chunk1.csv.fam, /path/to/chr1_chunk1.csv.log]` and so on...

I want to split the path such as I get:
`[chr1, /path/to/chr1_chunk1]`

Using the following command I can get such output:
`out_ch_from_previous_process.map{chr,bfiles -> tuple(chr,bfiles[0].toString().split('\\.')[0])}.set{in_for_next_process}
`

In the next process:

```
process next_pros1 {
input:
tuple val(chr),path(bfiles) from in_for_next_process
output:
tuple ("${chr}"),path("${chr}*") into in_for_following
script:
"""
plink --bfile ${bfiles} --maf 0.05 --make-bed --out ${bfiles}_maf
"""
}

```
It gives me error:
``There is no such file chr1_chunk1.csv.fam
``
Did I explain my situation correctly? How could I get it right?

**Comment by complexgenome** [comment]
@zillurbmb51  Hi,
were you able to find solution to it?

**Reply by mribeirodantas** [reply]
Are you using DSL1? If you're using DSL2, please open a new issue explaining exactly your situation and I'll try to help you 😉
